addons/mail/models/mail_template.py

```diff
@@ -214,7 +214,7 @@ def generate_email(self, res_ids, fields):
                         result, format = res
 
                     # TODO in trunk, change return format to binary to match message_post expected format
-                    result = base64.b64encode(result.encode())
+                    result = base64.b64encode(result)
```

Comment:
Surely this is going to blow up when the `else` branch above is taken, as in that case `_render` will return a `Markup`/`str`, which has to be explicitly encoded?

Alternatively update `_render_template` so `_render_qweb_html` and `_render_qweb_text` consistently return bytes again (this will probably require reverting other changes).

Reply:
Note that currently this is broken. I had issues in current master with templates and reports, it was crashing lamentablement.

Reply:
That's what this PR tries to fix innit?

Reply:
> Note that currently this is broken. I had issues in current master with templates and reports, it was crashing lamentablement.

Indeed, just clicking on "Send to client" on a sale order will raise an error.

> Alternatively update _render_template so _render_qweb_html and _render_qweb_text consistently return bytes again (this will probably require reverting other changes).

Done it like you suggested